src/main/java/edu/tamu/sage/service/SolrDiscoveryService.java

```diff
@@ -271,4 +282,26 @@ private ResultSet querySolrCore(DiscoveryView discoveryView, SolrQuery query, Se
         return new ResultSet(results, facetFilters);
     }
 
+    private void processResult(SolrDocument solrDoc) {
+        solrDoc.getFieldNames().forEach(name -> {
+
+            Object value = solrDoc.getFieldValue(name);
```

Comment:
Have you tried `getFieldValues`? https://lucene.apache.org/solr/8_5_2/solr-solrj/org/apache/solr/common/SolrDocument.html#getFieldValues-java.lang.String-

Will always return a Collection and will not require the conditions. I am not sure if this function will return on non-repeatable fields. If it does, I suggest using it instead.

Reply:
Thanks for catching that. I had tried to use that method on the aggregation side last sprint, but it didn't work out then. I didn't think to try it again for this one.